MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -0,0 +1,241 @@
+        modifiedEventDictionary["sdkVersion"] = sdkVersion
+        
+        modifiedEventDictionary["eventVersion"] = 2
+        
+        modifiedEventDictionary["profile"] = profile
+        modifiedEventDictionary["simulation"] = simulation
+        
+        modifiedEventDictionary["sessionIdentifier"] = sessionIdentifier
+        modifiedEventDictionary["originalRequestIdentifier"] = originalRequestIdentifier
+        modifiedEventDictionary["requestIdentifier"] = requestIdentifier
+        
+        modifiedEventDictionary["lat"] = lat
+        modifiedEventDictionary["lng"] = lng
+        
+        modifiedEventDictionary["originalGeometry"] = originalGeometry
+        modifiedEventDictionary["originalEstimatedDistance"] = originalEstimatedDistance
+        modifiedEventDictionary["originalEstimatedDuration"] = originalEstimatedDuration
+        
+        modifiedEventDictionary["geometry"] = geometry
+        modifiedEventDictionary["estimatedDistance"] = estimatedDistance
+        modifiedEventDictionary["estimatedDuration"] = estimatedDuration
+        
+        modifiedEventDictionary["distanceCompleted"] = distanceCompleted
+        modifiedEventDictionary["distanceRemaining"] = distanceRemaining
+        modifiedEventDictionary["durationRemaining"] = durationRemaining
+        
+        modifiedEventDictionary["rerouteCount"] = rerouteCount
+        
+        modifiedEventDictionary["volumeLevel"] = volumeLevel
+        modifiedEventDictionary["screenBrightness"] = screenBrightness
+        
+        modifiedEventDictionary["batteryPluggedIn"] = batteryPluggedIn
+        modifiedEventDictionary["batteryLevel"] = batteryLevel
+        modifiedEventDictionary["applicationState"] = applicationState
+        
+        return modifiedEventDictionary
+    }
+}
+
+extension MMEEventsManager {
+    func addDefaultEvents(routeController: RouteController) -> [String: Any] {
+        return DefaultEventDictionary(routeController: routeController, session: routeController.sessionState).convertToDictionary()
+    }
+}
+
+extension UIApplicationState {
+    var telemetryString: String {
+        get {
+            switch self {
+            case .active:
+                return "Foreground"
+            case .inactive:
+                return "Inactive"
+            case .background:
+                return "Background"
+            }
+        }
+    }
+}
+
+extension UIDevice {
+    @nonobjc var machine: String {
+        get {
+            var systemInfo = utsname()
+            uname(&systemInfo)
+            let machineMirror = Mirror(reflecting: systemInfo.machine)
+            let identifier = machineMirror.children.reduce("") { identifier, element in
+                guard let value = element.value as? Int8, value != 0 else { return identifier }
+                return identifier + String(UnicodeScalar(UInt8(value)))
+            }
+            
+            return identifier
+        }
+    }
+}
+
+extension CLLocation {
```

Comment:
We have an extension `CLLocation.swift` in `MapboxCoreNavigationTests`, can we consolidate them?
CLLocation <-> dictionary would be nice to provide in core.